qualtran/surface_code/magic_state_factory.py

```diff
@@ -36,3 +36,7 @@ def n_cycles(self, n_magic: AlgorithmSummary) -> int:
     @abc.abstractmethod
     def distillation_error(self, n_magic: AlgorithmSummary, phys_err: float) -> float:
         """The total error expected from distilling magic states with a given physical error rate."""
+
+    @abc.abstractmethod
+    def spacetime_footprint(self) -> float:
+        """The spacetime cost of producing a single magic state."""
```

Comment:
I think this needs more documentation. What are the units? logical or physical? There are existing properties named `footprint` and `n_cycles`. Is this just the product of those two? 

Should this be derived from those other two methods? What happens if they're inconsistent?

Reply:
bump

Reply:
done